### ISSUE TITLE
Filter by private shows 4 but finds no resources

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#1135]
In alpha pre-release deploy when not logged in, I can Select Discover, then expand Filter by availability, then it shows 4 next to private.  This is incorrect as since not logged in there should be no private resources accessible.  If I select the private box 4 resources are shown.  Clicking on each of them shows that they are actually "Discoverable"

**Comment by xuezhaokun** [comment]
@dtarb Should I just remove private option from there? 
@alvacouch Which is the correct way to index "discoverable" property?
```
    def prepare_discoverable(self, obj):
        if(obj.raccess.public | obj.raccess.discoverable):
            return True
        else:
            return False
```

or 
```

    def prepare_discoverable(self, obj):
        return obj.can_be_public_or_discoverable;
```

**Comment by pkdash** [comment]
I don't think the 2nd option should be used


Sent from Outlook Mobile






On Wed, Apr 20, 2016 at 5:06 PM -0700, "Zhaokun Xue" <notifications@github.com> wrote:





@dtarb Should I just remove private option from there?
@alvacouch Which is the correct way to index "discoverable" property?
```
    def prepare_discoverable(self, obj):
        if(obj.raccess.public | obj.raccess.discoverable):
            return True
        else:
            return False
```

or
```

    def prepare_discoverable(self, obj):
        return obj.can_be_public_or_discoverable;
```

---

**Comment by xuezhaokun** [comment]
@pkdash Thanks! @dtarb Those four private resources are discoverable. Should I just show all public resources instead of all discoverable resources?

**Comment by dtarb** [comment]
There is a discoverable category and those 4 resources should be part of that group.  They should not appear with the private group.

Private resources should only show if a user is logged in and if the user is owner or has been granted view or edit permissions.  If this is not feasible, I am OK with private resources not showing, as they appear on MyResources.

I am not sure what you are asking about showing all public resources instead of all discoverable resources.  It depends on which check boxes are checked.  If the public check box is checked, show public, if discoverable checked show discoverable, if both checked, show both.

**Comment by xuezhaokun** [comment]
@dtarb I am going to hide private option in availability.

**Comment by horsburgh** [comment]
I vote for simplifying if we can. The discovery page is for users to find out about what others are doing. If they want to see their own resources (and in particular their own private, non-discoverable resources), that's what the My Resources page is for. 

I don't think private, non-discoverable resources should show up at all in searches on the discovery page (regardless of whether you are logged in or not).

I also don't think that "Discoverable" should even be a filter.  Anything that shows up in a search result is by definition discoverable.  Nothing that isn't discoverable should ever show up in the discovery page - whether you are logged in or not.

**Comment by dtarb** [comment]
Fixed in 1.6.10 Candidate release.
